driver-core/src/main/com/mongodb/MongoClientSettings.java

```diff
@@ -517,6 +523,38 @@ public Builder autoEncryptionSettings(@Nullable final AutoEncryptionSettings aut
             return this;
         }
 
+        /**
+         * Sets the time limit for the full execution of an operation.
+         *
+         * <ul>
+         *   <li>{@code null} means that the timeout mechanism will defer to using:
+         *    <ul>
+         *        <li>{@code serverSelectionTimeoutMS}: How long the driver will wait for server selection to succeed before throwing an
+         *        exception.</li>
+         *        <li>{@code waitQueueTimeoutMS}: The maximum wait time in milliseconds that a thread may wait for a connection to become
+         *        available</li>
+         *        <li>{@code socketTimeoutMS}: How long a send or receive on a socket can take before timing out.</li>
+         *        <li>{@code wTimeoutMS}: How long the server will wait for the write concern to be fulfilled before timing out.</li>
+         *        <li>{@code maxTimeMS}: The cumulative time limit for processing operations on a cursor.</li>
+         *        <li>{@code maxCommitTimeMS}: The maximum amount of time to allow a single {@code commitTransaction} command to execute.
+         *   </ul>
+         *   </li>
+         *   <li>{@code 0} means infinite timeout.</li>
+         *    <li>{@code > 0} The time limit to use for the full execution of an operation.</li>
+         * </ul>
+         *
+         * @param timeout the timeout
+         * @param timeUnit the time unit
+         * @return this
+         * @since 4.x
+         * @see #getTimeout
+         */
+        public Builder timeout(final long timeout, final TimeUnit timeUnit) {
```

Comment:
I did not review documentation for methods similar to `ConnectionString.getTimeout`. I propose to write this documentation once on a method and link to that method from other places used for setting/getting the same option. If we not do that and instead have copies of the documentation in multiple places, we will likely end up having inconsistent documentation.

Reply:
@jyemin any thoughts on this? I agree with @stIncMale it would reduce the number copy pasta errors as well.

Perhaps just once in each class is a good compromise - the reader of the API docs won't jump around to much.

Reply:
I think we should be consistent with current practice.  It would be weird if this were the only option for which we do it differently.  

I would support a future project where we consolidate all configuration documentation in MongoClientSettings. It's always been annoying to have do duplicate it in several places.

Reply:
Ack